### PR TITLE
More useful methods for user management

### PR DESCRIPTION
**Pull request by Honigeintopf**
Issue: https://github.com/metal-stack-cloud/api-server/issues/444

API for the following are done:

- Admins Project List
- Admin tenant member add
- Admin tenant member remove
- Admin tenant remove
- Admin Project remove

### CODE REVIEW

**Comment by Honigeintopf** [comment]
Should I have better responses on success, so for example when deleting a member from the tenant?

I think it's sufficient like this, because admin already knows what he is deleting, seeing what you deleted after you input what you want to delete is not necessary.

File: proto/admin/v1/project.proto [hunk 1]
```diff
@@ -44,4 +44,6 @@ message ProjectServiceRemoveProjectRequest {
 message ProjectServiceRemoveProjectResponse {
   // Success is a flag indicating if the member was removed successfully
   bool success = 1;
+  // Optional message to provide additional details
+  optional string message = 2;
```
Comment (majst01) [inline proto/admin/v1/project.proto:48]: Isn't the error message descriptive enough ?

Comment (Honigeintopf) [reply proto/admin/v1/project.proto:48]: I'm using this for success return in the api-server to show in the CLI, should I remove it?

```
message := fmt.Sprintf("Member %s successfully removed from project %s", req.MemberId, req.ProjectId)
	response := &adminv1.ProjectServiceRemoveProjectResponse{
		Success: true,
		Message: &message,
	}
```

**Comment by Honigeintopf** [comment]
Removed Do not Merge, Label

File: proto/admin/v1/tenant.proto [hunk 2]
```diff
@@ -79,3 +83,19 @@ message TenantServiceRevokeResponse {
...
+  // Email of the member to add
+  string member_id = 2;
+  // Role to assign to the new member
+  api.v1.TenantRole role = 3;
+}
+
+// TenantServiceAddMemberResponse is the response payload for the add member request
+message TenantServiceAddMemberResponse {
+  // Confirmation of the addition
+  bool success = 1;
```
Comment (Gerrit91) [inline proto/admin/v1/tenant.proto:100]: I think we can assume success already from the status code of the response. Can be left empty. Or is there another reason this is necessary?

Comment (majst01) [reply proto/admin/v1/tenant.proto:100]: @Honigeintopf please remove the bool flag

File: proto/admin/v1/tenant.proto [hunk 3]
```diff
@@ -79,3 +83,19 @@ message TenantServiceRevokeResponse {
...
 }
+
+// TenantServiceAddMemberRequest is the request payload for adding a member to a tenant
+message TenantServiceAddMemberRequest {
+  // Tenant ID to which the member will be added
+  string tenant_id = 1;
+  // Email of the member to add
+  string member_id = 2;
+  // Role to assign to the new member
+  api.v1.TenantRole role = 3;
```
Comment (majst01) [inline proto/admin/v1/tenant.proto:94]: At least add the fore-mentioned validation even if it is not working yet.

**Review by majst01 — CHANGES_REQUESTED** [review]

File: proto/api/v1/tenant.proto [hunk 4]
```diff
@@ -125,7 +125,9 @@ message TenantMember {
   // Id is the user id of the member
   string id = 1;
   // Role is the role of the member
-  TenantRole role = 2;
+  TenantRole role = 2 [(validate.rules).enum = {
+    not_in: [0]
```
Comment (majst01) [inline proto/api/v1/tenant.proto:129]: sorry for complaining again, please add this validation to the new Service Request TenantServiceAddMemberRequest

Comment (majst01) [reply proto/api/v1/tenant.proto:129]: Better would be:

```
[(validate.rules).enum.defined_only = true];

```

**Review by majst01 — CHANGES_REQUESTED** [review]

File: proto/api/v1/tenant.proto (hunk 4 above)
Comment (Honigeintopf) [reply proto/api/v1/tenant.proto]: Okay will remove out of the API, the validation.
Only validation will be in the new admin function

**Pull request merged**